simplesat/sat/policy/undetermined_clause_policy.py

```diff
@@ -89,13 +93,12 @@ def get_next_package_id(self, assignments, clauses):
         assert assignments.get(candidate_id) is None, \
             "Trying to assign to a variable which is already assigned."
 
-        if not self.prefer_installed:
-            # If this exact package version is available locally, use it.
-            # NOTE: when repository priority is implemented, this will cause
-            # the virtual <installed> repository to act as if it always has the
-            # highest priority.
-            key = self._package_key(candidate_id)
-            candidate_id = self._preferred_package_ids.get(key, candidate_id)
+        # If this exact package version is available locally, use it.
+        # NOTE: when repository priority is implemented, this will cause
+        # the virtual <installed> repository to act as if it always has the
+        # highest priority.
+        key = self._package_key(candidate_id)
```

Comment:
Previously, if `prefer_installed` were `True`, we would default to suggesting the `package_id` of installed packages, which come from the `installed_repository`. This block made sure that if we were _not_ using `prefer_installed` we would still get the package from the installed repo rather than the remote one--it prevented an unnecessary `remove`, `install` step to get the same version of a package.

With the new changes, we always call this block to make sure that we use the installed repo if possible. There is no change in behavior. We could put this into a `if candidate_id not in `self.prefer_installed_pkg_ids` block, but that seemed like an extra bit of logic that doesn't gain us much in terms of speed